src/transforms/rename.jl

```diff
@@ -0,0 +1,37 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+  Rename(:col₁ => :newcol₁, :col₂ => :newcol₂, ..., :col₁ => :newcolₙ)
+
+The transform that renames `col₁` to `newcol₁`, `col₂` to `newcol₂`, ...
+"""
+struct Rename <: Stateless
+  names::Dict{Symbol,Symbol}
+end
+
+function apply(transform::Rename, table)
+  oldnames = Tables.columnnames(table)
+  newnames = map(oldnames) do oldname
+    if oldname in keys(transform.names) 
+      transform.names[oldname]
+    else
+      oldname
+    end
+  end
+  acols = [i for i in Tables.columns(table)]
+  𝒯 = (; zip(newnames, acols)...) |> Tables.materializer(table)
+  𝒯, nothing
+end
+
+function revert(transform::Rename, table)
+  # reversing the key-value pairs of the Dict
+  new_names = Dict()
+  for (new, old) in transform.names
+    new_names[old] = new
+  end
+
+  reversed_transform = Rename(new_names)
+  apply(reversed_transform, table) |> first
```

Comment:
We don't want create a new transform to revert. What we need to do is save the old names in the cache and use the cache to revert the operation.

Reply:
Alternatively you can refactor this common set of instructions into an separate function `_rename` that does the actual work, and use this function in multiple places. This will facilitate future rearrangements of the code, and is good software engineering practice.

Reply:
I am gonna do it this second way, because I am not sure if there is anything useful to put in that cache.

So it is ok to: 
- refactor that `_rename`, 
- call it twice instead of creating another transform, 
- and not use `cache`. (pass it around as `nothing` instead)